ibis/sql/postgres/udf/api.py

```diff
@@ -0,0 +1,218 @@
+import inspect
+from decimal import Decimal
+import collections
+import itertools
+
+import sqlalchemy
+
+import ibis.expr.rules as rlz
+from ibis.expr import datatypes
+from ibis.expr.signature import Argument as Arg
+from ibis.sql.postgres.compiler import PostgresUDFNode, add_operation
+
+
+_udf_name_cache = collections.defaultdict(itertools.count)
+
+
+# type mapping based on: https://www.postgresql.org/docs/10/plpython-data.html
+sql_default_type = 'VARCHAR'
+
+pytype_sql = {
```

Comment:
I think we'd probably want to use `sqlalchemy.types` here, to depend on the 10+ years of work spent making sure that its types compile to the correct postgres type.

Reply:
I switched to using `ibis` and `sqlalchemy`s builtin mechanisms for translating types.